The test case is known broken upstream.

See: https://gitlab.gnome.org/GNOME/libsoup/-/issues/352

diff -pur libsoup-3.4.2.orig/tests/meson.build libsoup-3.4.2/tests/meson.build
--- libsoup-3.4.2.orig/tests/meson.build	2023-10-22 19:00:24.734407553 +0200
+++ libsoup-3.4.2/tests/meson.build	2023-10-22 19:00:30.961092172 +0200
@@ -95,11 +95,6 @@ tests = [
   {'name': 'server-auth'},
   {'name': 'server'},
   {'name': 'sniffing'},
-  {'name': 'ssl',
-   'dependencies': [gnutls_dep],
-   'depends': mock_pkcs11_module,
-   'c_args': '-DHAVE_GNUTLS=@0@'.format(gnutls_dep.found() ? 1 : 0),
-  },
   {'name': 'streaming'},
   {'name': 'timeout'},
   {'name': 'tld'},
